src/components/wallets/WalletTransactionList.tsx

```diff
@@ -0,0 +1,244 @@
+import { useEffect } from 'react'
+import styled from 'styled-components'
+import { DateTime } from 'luxon'
+import { gql } from '@apollo/client'
+
+import { theme } from '~/styles'
+import { GenericPlaceholder } from '~/components/GenericPlaceholder'
+import ErrorImage from '~/public/images/maneki/error.svg'
+import EmptyImage from '~/public/images/maneki/empty.svg'
+import { Avatar, Button, Icon, Skeleton, Typography } from '~/components/designSystem'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import {
+  useGetWalletTransactionsLazyQuery,
+  Wallet,
+  WalletTransactionStatusEnum,
+  WalletTransactionTransactionTypeEnum,
+} from '~/generated/graphql'
+import { intlFormatNumber } from '~/core/intlFormatNumber'
+
+gql`
+  query getWalletTransactions($walletId: ID!, $page: Int, $limit: Int) {
+    walletTransactions(walletId: $walletId, page: $page, limit: $limit) {
+      metadata {
+        currentPage
+        totalPages
+        totalCount
+      }
+      collection {
+        id
+        status
+        amount
+        creditAmount
+        settledAt
+      }
+    }
+  }
+
+  fragment WalletInfosForTransactions on Wallet {
```

Comment:
You should always have an id on a fragment

Reply:
Indeed make sense, udpated